server/handlers.go

```diff
@@ -449,53 +459,54 @@ func (s *Server) putHandler(w http.ResponseWriter, r *http.Request) {
 
 	contentLength := r.ContentLength
 
-	var reader io.Reader
-
-	reader = r.Body
-
 	defer r.Body.Close()
 
-	if contentLength == -1 {
-		// queue file to disk, because s3 needs content length
-		var err error
-		var f io.Reader
-
-		f = reader
-
-		var b bytes.Buffer
+	var err error
+	var file *os.File
+	defer s.cleanTmpFile(file)
```

Comment:
Lets move this after the actual creation/assignment in L468. On my tests the files didnt get removed for both failures and successes.

Reply:
good catch!